### PR TITLE
vsx-registry: display incompatible extensions

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

Fixes: https://github.com/eclipse-theia/theia/issues/9676.

The commit adds the functionality to display incompatible extensions when performing a search. The changes will now display incompatible extensions but mark them as disabled, and non-installable. The behavior is also controllable by a new preference so users can decide if they want the potential extra noise when searching.

_Example: using an api-version of 1.30.0_:

incompatible-plugins

#### How to test

1. start the application
2. open the extensions-view
3. perform a search - if an extension has no compatible version then it should be marked as disabled and non-installable
  3.1. hovering over the disabled extension should correctly display the tooltip
  3.2 the plugin should not resolve, the readme won't be displayed
4. updating the preference `'extensions.displayIncompatible'` and re-searching should not yield incompatible results
5. repeat the steps by starting with `yarn start --vscode-api-version=1.30.0` for additional incompatible results

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

File: packages/vsx-registry/src/browser/vsx-extension.tsx [hunk 1]
```diff
@@ -414,16 +421,16 @@ const downloadCompactFormatter = new Intl.NumberFormat(undefined, { notation: 'c
...
-        return <div className='theia-vsx-extension noselect' data-for={tooltipId} data-tip={tooltip}>
+        return <div className={`theia-vsx-extension noselect ${compatible ? '' : 'incompatible'}`} data-for={tooltipId} data-tip={tooltip}>
             {iconUrl ?
                 <img className='theia-vsx-extension-icon' src={iconUrl} /> :
                 <div className='theia-vsx-extension-icon placeholder' />}
             <div className='theia-vsx-extension-content'>
                 <div className='title'>
                     <div className='noWrapInfo'>
-                        <span className='name'>{displayName}</span> <span className='version'>{version}</span>
+                        <span className='name'>{displayName}</span> <span className='version'>{version !== undefined ? version : 'No Compatible Version'}</span>
```
Comment (msujew) [inline packages/vsx-registry/src/browser/vsx-extension.tsx:433]: Suggestion: please use the nls.localize function for the `No Compatible Version` string.

**Review by msujew — COMMENTED** [review]
The behavior looks good to me:

- [x] Incompatible version (tested this with API version 1.53.2 and 1.30.0) are shown but not installable.
- [x] Switching the preference on/off correctly starts/stops displaying extensions with incompatible versions
- [x] The visual changes work well with different themes.

I have a few suggestions regarding i18n though.

File: packages/vsx-registry/src/browser/vsx-extension.tsx [hunk 2]
```diff
@@ -398,7 +403,9 @@ export abstract class AbstractVSXExtensionComponent extends React.Component<Abst
             return <div><button className="theia-button action" onClick={this.uninstall}>Uninstall</button>
                 <div className="codicon codicon-settings-gear action" onClick={this.manage}></div></div>;
         }
-        return <button className="theia-button prominent action" onClick={this.install}>Install</button>;
```
Comment (vince-fugnitto) [inline packages/vsx-registry/src/browser/vsx-extension.tsx:401]: @msujew your previous comment (https://github.com/eclipse-theia/theia/pull/10424/files#r751129871) reminds me, shouldn't we localize the buttons?

Comment (msujew) [reply packages/vsx-registry/src/browser/vsx-extension.tsx:401]: We should, yeah. `nls.localizeByDefault` should do the trick for both of them 👍

**Review by colin-grant-work — COMMENTED** [review]
A few comments:
### Functionality

- If you search for plugins and then change the preference, the current search result set doesn't change (the search isn't rerun). Particularly when moving from not showing -> showing, I think it would be better if the display did update.
- It seems that if a plugin is incompatible, you can't open an extension view widget for it to read its readme. That was unexpected, and seems undesirable.

### Experience

Displaying a plugin that you can't interact with at all seems odd. Certainly it should be possible to open the extension view for it to see the readme, but I think it might also make sense to allow the user to install an incompatible plugin after a warning dialog that points out that the plugin may not work. Perhaps beyond the scope of this PR, but an excellent addition to it, if you're willing, would be to take the logic from #9330 to give the user the chance to select the oldest available version to install, in case that works better than the latest.

**Comment by vince-fugnitto** [comment]
Converted the pull-request to a draft in favor of completing #9330 first to get a better behavior in this pull-request :+1:

**Comment by vince-fugnitto** [comment]
Moved the draft to my fork in the meantime https://github.com/vince-fugnitto/theia/pull/9. 
The changes should be revisited once the following are completed:
- https://github.com/eclipse-theia/theia/pull/9330
- https://github.com/eclipse-theia/theia/pull/11084

**Pull request closed without merging**